packages/flutter/test/material/text_field_test.dart

```diff
@@ -3341,4 +3341,38 @@ void main() {
     expect(minOffset, 0.0);
     expect(maxOffset, 50.0);
   });
+
+  testWidgets('TextField style is merged with theme', (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/23994
+
+    final ThemeData themeData = ThemeData(
+      textTheme: TextTheme(
+        subhead: TextStyle(
+          color: Colors.blue[500],
+        ),
+      ),
+    );
+
+    Widget buildFrame() {
+      return MaterialApp(
+        theme: themeData,
+        home: Material(
+          child: Center(
+            child: Builder(
+              builder: (BuildContext context) {
+                return TextField(
+                  style: const TextStyle(),
+                );
+              },
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildFrame());
+    await tester.pumpAndSettle();
+    final EditableText editableText = tester.widget(find.byType(EditableText));
+    expect(editableText.style.color, themeData.textTheme.subhead.color);
```

Comment:
TextStyle overrides `operator ==` so you can compare editableText.style with ...subhead

Reply:
This seems to fail when I try it.  It looks like the editableText style is having some properties changed somewhere along the way so it's no longer equal to the theme.

Reply:
That's probably because the theme's TextStyles are localized when the app's theme looked up with `ThemeData.of()'.  See https://docs.flutter.io/flutter/material/ThemeData/localize.html and https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/theme.dart#L137

That's why the value of `Theme.of(context).textTheme.subhead` isn't the same as the textstyle created by the test and to the MaterialApp's theme (ThemeData) parameter.